crates/rpc/rpc/src/eth/api/transactions.rs

```diff
@@ -1288,6 +1288,20 @@ where
 
         Ok(None)
     }
+
+    pub(crate) async fn raw_transaction_by_block_and_tx_index(
+        &self,
+        block_id: impl Into<BlockId>,
+        index: Index,
+    ) -> EthResult<Option<Bytes>> {
+        if let Some(block) = self.block_with_senders(block_id.into()).await? {
+            if let Some(tx) = block.into_transactions_ecrecovered().nth(index.into()) {
```

Comment:
can access the vec directly because we don't need the recovered variant
```suggestion
            if let Some(tx) = block.transactions().nth(index.into()) {
```